new-frontiers-14/frontier-station-14
# Proper Fuel-Grade Conversion, Materials & Sprites

## About the PR



Creates materials for fuel-grade plasma, uranium and bananium.  Creates a new component and system that converts all materials in a generator from a given input material (generally "quality" material) to a given output (generally "fuel-grade" material).  Currently only works 1:1.

All fuel inserted into a portable generator is converted 1:1 into fuel-grade material, which each generator now runs on.  The ejection or deconstruction of the material just removes this fuel-grade fuel out of the machine.

Adds the fuel-grade material sprites given by GhostPrice (many thanks).

## Why / Balance


Fixes #2554.

This solution seems cleaner than what existed before.

## How to test


1. Spawn the shuttle versions of the PACMAN, SUPERPACMAN and HYPERPACMAN.  They should all be running.
2. Insert regular, proper sheets of quality fuel into each.  They should all take the fuel and adjust their amounts properly.
3. Insert fuel-grade fuel into each, make sure you have enough space for each.
4. Eject the material from each generator, it should all be fuel-grade.

## Media


Three generators, their deconstructed states below (each with the fuel they discarded on top of them), and the fuel-grade material that can be used to fuel them underneath.
![image](https://github.com/user-attachments/assets/88900117-d924-4e78-a938-25cc23905274)

## Requirements



## Breaking changes


**Changelog**


:cl:
- fix: Deconstructing a portable generator with fuel now returns fuel-grade material.
- add: Fuel-grade materials now have sprites from GhostPrince.
